### PR TITLE
Delete all chats associated with calling user

### PR DESCRIPTION
**Pull request by jdtoombs**
### Motivation and Context
Allow user to easily clear chats, instead of one by one.

Just doing some final load testing; however, code should be ready for review. Have a few issues I want to highlight. See comments below.

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [Contribution Guidelines](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: webapi/Storage/ChatSessionRepository.cs [hunk 1]
```diff
@@ -35,4 +35,14 @@ public Task<IEnumerable<ChatSession>> QueryEntitiesAsync(Expression<Func<ChatSes
     {
         return base.StorageContext.QueryEntitiesAsync(predicate);
     }
+
+    /// <summary>
+    /// Retrieves a list of chat sessions associated to userId.
+    /// </summary>
+    /// <param name="userId">userId to fetch chat sessions.</param>
+    /// <returns>A list of ChatSessions.</returns>
+    public Task<IEnumerable<ChatSession>> FindByUserIdAsync(string userId)
```
Comment (jdtoombs) [inline webapi/Storage/ChatSessionRepository.cs:44]: The issue with this is that existing chats don't have a userId associated with them, but the new ones do as I have added this property.. wondering if I need to update the existing chats?

File: webapi/Controllers/ChatHistoryController.cs [hunk 2]
```diff
@@ -351,6 +353,46 @@ public async Task<ActionResult<IEnumerable<MemorySource>>> GetSourcesAsync(Guid
...
+    )
+    {
+        IEnumerable<ChatSession> sessions = await this._sessionRepository.FindByUserIdAsync(this._authInfo.UserId);
+
+        var deleteTasks = sessions.Select(async session =>
+        {
+            try
+            {
+                await Task.WhenAll(
+                    this._messageRepository.DeleteAllFromPartitionAsync(session.Id),
```
Comment (jdtoombs) [inline webapi/Controllers/ChatHistoryController.cs:377]: Deleting from partition seems to be much more efficient than the `DeleteChatResources` method that iterates through each message etc

File: webapi/Storage/VolatileContext.cs [hunk 3]
```diff
@@ -67,6 +67,22 @@ public Task DeleteAsync(T entity)
         return Task.CompletedTask;
     }
 
+    /// <inheritdoc/>
```
Comment (jdtoombs) [inline webapi/Storage/VolatileContext.cs:70]: Need to test this

**Comment by mattthiessendev** [comment]
Thought: This PR assumes we will have one user associated with one chat session. This assumption goes against a lot of the infrastructure we inherited from Microsoft when we forked this project. We should verify we are not going to support multi-user chats in the future. Otherwise we will have a painful refactor to take on later.

**Comment by jdtoombs** [comment]
> Thought: This PR assumes we will have one user associated with one chat session. This assumption goes against a lot of the infrastructure we inherited from Microsoft when we forked this project. We should verify we are not going to support multi-user chats in the future. Otherwise we will have a painful refactor to take on later.

Good call, will raise at stand up tomorrow

**Comment by wdavidcamden** [comment]
After standup, i believe the approach is now to delete the participant from the chat session and leave the chat session alone. Followup task to create a process to cleanup orphaned chats.

**Pull request closed without merging**